### DISCUSSION TITLE
Adding my own xarray datacube w/ time slider, is this currently possible?

### DISCUSSION THREAD

**Discussion by lalligagger** [opengeos/leafmap#451] [Q&A]
Hi Qiusheng, this is such a cool project! And I'm still curious about `anymap` for 3D viz once I have time to put some attention on that. Thanks for adding me to that repo.

Onto the issue topic: I will definitely be using your STAC+timeslider example and also have some custom data to have, currently in xarray ds/ da's.

1. What would be your suggestion to load custom datasets? I'm using `odc` to load the "raw" data from a STAC search,  just doing basic index calcs so far, and expect resolution changes, etc. so more metadata changes than just adding new band.
2. I've gotten through the xarray -> DIY STAC catalog process before, but would prefer to skip it. I was considering using `xpublish` and seeing if I could hook this into `leafmap`. Can you give any guidance on if this could work or how else you might approach? The publishing and viewing would all happen on the same machine, so I am looking for lightweight over performant solutions, and no real need for persistent caching.
3. Ultimately I would love to have the raw & processed data compared with a swiper and time scrubber. Assuming this would take some work, if you have any pointers I might give it a try!

Thanks very much.

**Comment by lalligagger** [comment]
Could a time key be added to your Titiler example, possibly?

**Comment by giswqs** [comment]
The time slider needs tile layers, whether remote tiles through TiTler or local tiles through localtileserver. I guess you are referring to local tiles. For now, localtileserver does not support in-memory raster yet. Once it does, you should be able to use it the generate tiles from your own xarray datacube. See https://github.com/banesullivan/localtileserver/issues/147
